File: src/main/java/dev/paigewatson/layoutmaster/client/services/CarTypeService.java
package dev.paigewatson.layoutmaster.client.services;

import dev.paigewatson.layoutmaster.models.goods.GoodsType;
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation;
import dev.paigewatson.layoutmaster.models.rollingstock.CarType;

import java.util.List;

public interface CarTypeService
{
    List<AARDesignation> allAARDesignations();

    CarType saveCarTypeToDatabase(CarType carTypeToSave);

    List<CarType> allCarTypes();

    CarType carTypeForAAR(AARDesignation expectedAARDesignation);

    List<CarType> carTypesThatCarryGoodsType(GoodsType expectedGoodsType);
}

File: src/main/kotlin/dev/paigewatson/layoutmaster/client/services/CarTypeService.kt
package dev.paigewatson.layoutmaster.client.services

import dev.paigewatson.layoutmaster.models.goods.GoodsType
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation
import dev.paigewatson.layoutmaster.models.rollingstock.CarType

interface CarTypeService {
    fun allAARDesignations(): List<AARDesignation>
    fun saveCarTypeToDatabase(carTypeToSave: CarType): CarType
    fun allCarTypes(): List<CarType>
    fun carTypeForAAR(expectedAARDesignation: AARDesignation): CarType
    fun carTypesThatCarryGoodsType(expectedGoodsType: GoodsType): List<CarType>
}
